src/analysisd/eventinfo.c

```diff
@@ -76,6 +76,10 @@ Eventinfo *Search_LastSids(Eventinfo *my_lf, RuleInfo *rule, __attribute__((unus
             goto end;
         }
 
+        if (!(rule->context_opts & GLOBAL_FREQUENCY) && (strcmp(lf->agent_id, my_lf->agent_id))){
```

Comment:
Check the existence of `lf->agent_id` and `my_lf->agent_id` to avoid a crash. The same for the other occurrences.